javascript/ql/src/experimental/heuristics/ql/src/Security/CWE-020/UntrustedDataToExternalAPI.ql

```diff
@@ -0,0 +1,22 @@
+/**
+ * @name Untrusted data passed to external API with additional heuristic sources
+ * @description Data provided remotely is used in this external API without sanitization, which could be a security risk.
+ * @id js/untrusted-data-to-external-api-more-sources
+ * @kind path-problem
+ * @precision low
+ * @problem.severity error
+ * @security-severity 7.8
+ * @tags experimental
+ *       security external/cwe/cwe-20
+ */
+
+import javascript
+import semmle.javascript.security.dataflow.ExternalAPIUsedWithUntrustedDataQuery
+import DataFlow::PathGraph
+import semmle.javascript.heuristics.AdditionalSources
+
+from Configuration config, DataFlow::PathNode source, DataFlow::PathNode sink
```

Comment:
I was wondering whether you could move the entire `select` statement to shared `query predicate problems()` . If so, you could write:
```ql
import javascript
import semmle.javascript.security.dataflow.ExternalAPIUsedWithUntrustedDataQuery as Query
import DataFlow::PathGraph
import semmle.javascript.heuristics.AdditionalSources

query predicate problems(DataFlow::PathNode a, DataFlow::PathNode b, DataFlow::PathNode c, string d, DataFlow::PathNode e, string f) {
  Query::problems(a,b,c,d,e,f) and b.getNode() instanceof HeuristicSource
}
```

Not sure if it is an improvement though.

Reply:
I don't think that's an improvement, so I prefer not to do that.  
I tried to keep the modifications to the ordinary queries as small as possible, so that they wouldn't be harder to read.  

`ConditionalBypass.ql` could use a `Query.qll` file though, so I'll get on that.